lib/MusicBrainz/Server/WebService/Validator.pm

```diff
@@ -224,8 +242,9 @@ sub validate_inc
         if (!exists $acc{$i} && !exists $extra{$i})
         {
             my @possible = grep {
-                my %all = map { $_ => 1 } @{ $extra_inc{$_} };
-                exists $all{$i}
+                my %all = map { $_ => 1 } @{ $extra_inc{$_}{generic} };
+                my %per_resource = map { $_ => 1 } @{ $extra_inc{$_}{$resource} };
```

Comment:
Like above, I think you will need the `// []` here to avoid the dreaded "Can't use an undefined value as an ARRAY reference", unless I'm missing a reason that wouldn't be hit here.